src/playlist-controller.js

```diff
@@ -2052,4 +2099,128 @@ export class PlaylistController extends videojs.EventTarget {
       videoDuration
     });
   }
+
+  pathwayAttribute_(playlist) {
+    return playlist.attributes['PATHWAY-ID'] || playlist.attributes.serviceLocation;
+  }
+  /**
+   * Initialize content steering listeners and apply the tag properties.
+   */
+  initContentSteeringController_() {
+    const initialMain = this.main();
+
+    if (!initialMain.contentSteering) {
+      return;
+    }
+
+    const updateSteeringValues = (main) => {
+      for (const playlist of main.playlists) {
+        this.contentSteeringController_.addAvailablePathway(this.pathwayAttribute_(playlist));
+      }
+
+      this.contentSteeringController_.assignTagProperties(main.uri, main.contentSteering);
+    };
+
+    updateSteeringValues(initialMain);
+
+    this.contentSteeringController_.on('content-steering', this.excludeThenChangePathway_.bind(this));
```

Comment:
Do we clear this event listener?

Reply:
I added an `off` call in the dispose function here: https://github.com/videojs/http-streaming/pull/1427/commits/e882de56db6a14be2e74b41ccd16ceba0e67cccf#diff-a167ea6828d7c01e63609400b5cb67b841a64432c629f7913fc04bc528fa3f3cR389

Reply:
Just want to clarify: Does it mean that when `dispose` is called - "contentSteeringController" is no longer usable and we don't need to re-init this listener?

Reply:
Correct. Dispose assumes we no longer want to use content steering functionality or that instance of a `contentSteeringController`.